From 3e57385db3a4f3b5b54a623dffeb69c3b9e85f9e Mon Sep 17 00:00:00 2001
From: Benjamin Berg <bberg@redhat.com>
Date: Thu, 1 Oct 2020 11:39:55 +0200
Subject: [PATCH] Use special exit code of 2 for unsupported platforms
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Exit with code 2 when thermald shuts down voluntarily because the
platform is unsupported. Only consider this error code as successful in
systemd unit.

This avoids leaving the unit in a non-failed state for fatal errors.
---
 data/thermald.service.in | 2 +-
 src/main.cpp             | 8 +++++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/data/thermald.service.in b/data/thermald.service.in
index f4416b0..49de893 100644
--- a/data/thermald.service.in
+++ b/data/thermald.service.in
@@ -4,7 +4,7 @@ ConditionVirtualization=no
 
 [Service]
 Type=dbus
-SuccessExitStatus=1
+SuccessExitStatus=2
 BusName=org.freedesktop.thermald
 ExecStart=@sbindir@/thermald --systemd --dbus-enable --adaptive
 
diff --git a/src/main.cpp b/src/main.cpp
index a5cff47..04b5c5c 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -51,6 +51,8 @@
 #define TD_DIST_VERSION PACKAGE_VERSION
 #endif
 
+#define EXIT_UNSUPPORTED 2
+
 extern int thd_dbus_server_init(void (*exit_handler)(int));
 
 // Lock file
@@ -351,7 +353,11 @@ int main(int argc, char *argv[]) {
 	if (ret != THD_SUCCESS) {
 		clean_up_lockfile();
 		closelog();
-		exit(EXIT_FAILURE);
+
+		if (ret == THD_ERROR)
+			exit(EXIT_UNSUPPORTED);
+		else
+			exit(EXIT_FAILURE);
 	}
 
 	// Start service requests on the D-Bus
-- 
2.27.0

